wagtail/api/v2/views.py

```diff
@@ -555,14 +555,18 @@ def get_queryset(self):
 
         # Allow pages to be filtered to a specific type
         try:
-            models = page_models_from_string(
-                request.GET.get("type", "wagtailcore.Page")
-            )
+            models_type = request.GET.get("type", None)
+            models = models_type and page_models_from_string(models_type) or []
         except (LookupError, ValueError):
             raise BadRequestError("type doesn't exist")
 
         if not models:
-            return self.get_base_queryset()
+            if self.model == Page:
+                return self.get_base_queryset()
+            else:
+                return self.model.objects.filter(
+                    id__in=self.get_base_queryset().values_list("id", flat=True)
```

Comment:
it is best to use `pk` instead of `id` as we should try to avoid making the assumption the primary key is "id"

Reply:
I've made changes accordingly. Can you please review them?